Build failures with mtl-2.3 and `transformers < 0.6`

```
src/Data/Constraint/Lifting.hs:402:70: error:
    • Could not deduce (MonadCont (ErrorT e a))
        arising from a use of ‘Dict’
      from the context: Error e
        bound by the instance declaration
        at src/Data/Constraint/Lifting.hs:402:10-48
      or from: MonadCont a
        bound by a type expected by the context:
                   MonadCont a => Dict (MonadCont (ErrorT e a))
        at src/Data/Constraint/Lifting.hs:402:70-73
    • In the first argument of ‘Sub’, namely ‘Dict’
      In the expression: Sub Dict
      In an equation for ‘lifting’: lifting = Sub Dict
    |
402 | instance Error e => Lifting MonadCont (ErrorT e) where lifting = Sub Dict
    |                                                                      ^^^^

src/Data/Constraint/Lifting.hs:403:77: error:
    • Could not deduce (MonadRWS r w s (ErrorT e a))
        arising from a use of ‘Dict’
      from the context: Error e
        bound by the instance declaration
        at src/Data/Constraint/Lifting.hs:403:10-55
      or from: MonadRWS r w s a
        bound by a type expected by the context:
                   MonadRWS r w s a => Dict (MonadRWS r w s (ErrorT e a))
        at src/Data/Constraint/Lifting.hs:403:77-80
    • In the first argument of ‘Sub’, namely ‘Dict’
      In the expression: Sub Dict
      In an equation for ‘lifting’: lifting = Sub Dict
    |
403 | instance Error e => Lifting (MonadRWS r w s) (ErrorT e) where lifting = Sub Dict
    |                                                                             ^^^^

src/Data/Constraint/Lifting.hs:404:76: error:
    • Could not deduce (MonadReader r (ErrorT e a))
        arising from a use of ‘Dict’
      from the context: Error e
        bound by the instance declaration
        at src/Data/Constraint/Lifting.hs:404:10-54
      or from: MonadReader r a
        bound by a type expected by the context:
                   MonadReader r a => Dict (MonadReader r (ErrorT e a))
        at src/Data/Constraint/Lifting.hs:404:76-79
    • In the first argument of ‘Sub’, namely ‘Dict’
      In the expression: Sub Dict
      In an equation for ‘lifting’: lifting = Sub Dict
    |
404 | instance Error e => Lifting (MonadReader r) (ErrorT e) where lifting = Sub Dict
    |                                                                            ^^^^

src/Data/Constraint/Lifting.hs:405:75: error:
    • Could not deduce (MonadState s (ErrorT e a))
        arising from a use of ‘Dict’
      from the context: Error e
        bound by the instance declaration
        at src/Data/Constraint/Lifting.hs:405:10-53
      or from: MonadState s a
        bound by a type expected by the context:
                   MonadState s a => Dict (MonadState s (ErrorT e a))
        at src/Data/Constraint/Lifting.hs:405:75-78
    • In the first argument of ‘Sub’, namely ‘Dict’
      In the expression: Sub Dict
      In an equation for ‘lifting’: lifting = Sub Dict
    |
405 | instance Error e => Lifting (MonadState s) (ErrorT e) where lifting = Sub Dict
    |                                                                           ^^^^

src/Data/Constraint/Lifting.hs:424:54: error:
    • Could not deduce (MonadCont (ListT a))
        arising from a use of ‘Dict’
      from the context: MonadCont a
        bound by a type expected by the context:
                   MonadCont a => Dict (MonadCont (ListT a))
        at src/Data/Constraint/Lifting.hs:424:54-57
    • In the first argument of ‘Sub’, namely ‘Dict’
      In the expression: Sub Dict
      In an equation for ‘lifting’: lifting = Sub Dict
    |
424 | instance Lifting MonadCont ListT where lifting = Sub Dict
    |                                                      ^^^^

src/Data/Constraint/Lifting.hs:425:59: error:
    • Could not deduce (MonadError e (ListT a))
        arising from a use of ‘Dict’
      from the context: MonadError e a
        bound by a type expected by the context:
                   MonadError e a => Dict (MonadError e (ListT a))
        at src/Data/Constraint/Lifting.hs:425:59-62
    • In the first argument of ‘Sub’, namely ‘Dict’
      In the expression: Sub Dict
      In an equation for ‘lifting’: lifting = Sub Dict
    |
425 | instance Lifting (MonadError e) ListT where lifting = Sub Dict
    |                                                           ^^^^

src/Data/Constraint/Lifting.hs:426:60: error:
    • Could not deduce (MonadReader r (ListT a))
        arising from a use of ‘Dict’
      from the context: MonadReader r a
        bound by a type expected by the context:
                   MonadReader r a => Dict (MonadReader r (ListT a))
        at src/Data/Constraint/Lifting.hs:426:60-63
    • In the first argument of ‘Sub’, namely ‘Dict’
      In the expression: Sub Dict
      In an equation for ‘lifting’: lifting = Sub Dict
    |
426 | instance Lifting (MonadReader r) ListT where lifting = Sub Dict
    |                                                            ^^^^

src/Data/Constraint/Lifting.hs:427:59: error:
    • Could not deduce (MonadState s (ListT a))
        arising from a use of ‘Dict’
      from the context: MonadState s a
        bound by a type expected by the context:
                   MonadState s a => Dict (MonadState s (ListT a))
        at src/Data/Constraint/Lifting.hs:427:59-62
    • In the first argument of ‘Sub’, namely ‘Dict’
      In the expression: Sub Dict
      In an equation for ‘lifting’: lifting = Sub Dict
    |
427 | instance Lifting (MonadState s) ListT where lifting = Sub Dict
    |                                                           ^^^^
cabal: Failed to build constraints-0.13.3.
```

--- a/CHANGELOG.markdown
+++ b/CHANGELOG.markdown
@@ -1,3 +1,8 @@
+next [????.??.??]
+-----------------
+* Correct the CPP introduced in `constraints-0.13.3` such that it works when
+  building with `mtl-2.3.*` or later combined with `transformers < 0.6`.
+
 0.13.3 [2022.01.31]
 -------------------
 * Allow building with `transformers-0.6.*` and `mtl-2.3.*`.

--- a/src/Data/Constraint/Lifting.hs
+++ b/src/Data/Constraint/Lifting.hs
@@ -399,10 +399,12 @@ instance (Show e, Show1 m) => Lifting Show (ErrorT e m) where lifting = Sub Dict
 instance (Eq e, Eq1 m) => Lifting Eq (ErrorT e m) where lifting = Sub Dict
 instance (Ord e, Ord1 m) => Lifting Ord (ErrorT e m) where lifting = Sub Dict
 instance (Read e, Read1 m) => Lifting Read (ErrorT e m) where lifting = Sub Dict
+# if !(MIN_VERSION_mtl(2,3,0))
 instance Error e => Lifting MonadCont (ErrorT e) where lifting = Sub Dict
 instance Error e => Lifting (MonadRWS r w s) (ErrorT e) where lifting = Sub Dict
 instance Error e => Lifting (MonadReader r) (ErrorT e) where lifting = Sub Dict
 instance Error e => Lifting (MonadState s) (ErrorT e) where lifting = Sub Dict
+# endif
 
 instance Lifting Functor ListT where lifting = Sub Dict
 instance Lifting Applicative ListT where lifting = Sub Dict
@@ -421,10 +423,12 @@ instance Show1 m => Lifting Show (ListT m) where lifting = Sub Dict
 instance Read1 m => Lifting Read (ListT m) where lifting = Sub Dict
 instance Ord1 m => Lifting Ord (ListT m) where lifting = Sub Dict
 instance Eq1 m => Lifting Eq (ListT m) where lifting = Sub Dict
+# if !(MIN_VERSION_mtl(2,3,0))
 instance Lifting MonadCont ListT where lifting = Sub Dict
 instance Lifting (MonadError e) ListT where lifting = Sub Dict
 instance Lifting (MonadReader r) ListT where lifting = Sub Dict
 instance Lifting (MonadState s) ListT where lifting = Sub Dict
+# endif
 #endif
 
 class Lifting2 p f where

